OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,375 @@
+        }
+
+        public RegistrationCode RegisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            castCombo.ClearSingles();
+            if (castCombo == null || castCombo.Combo == null)
+            {
+                _console.WriteLine("combination is null");
+                return RegistrationCode.InvalidCombination;
+            }
+            var combs = castCombo.Combo.ToLower().Split('/');
+            var combos = new List<long>();
+            foreach (string comstr in combs)
+            {
+                var hash = ParseCombination(comstr);
+                if (hash <= 0)
+                {
+                    return (RegistrationCode)hash;
+                }
+                combos.Add(hash);
+            }
+            foreach (long comb in combos)
+            {
+                _comboRegistry.Add(comb, combination);
+                if (comb < _maxUsefulKey)
+                {
+                    castCombo.AddSingle((KeyCode)comb);
+                }
+            }
+            _logger.Log("succesfully registered " + castCombo.Combo);
+            return RegistrationCode.AllNormal;
+        }
+
+        public RegistrationCode UnregisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            if (!_comboRegistry.ContainsValue(combination))
+            {
+                return RegistrationCode.InvalidCombination;
+            }
+            if (combination == null || castCombo.Combo == null)
+            {
+                _console.WriteLine("combination is null");
+                return RegistrationCode.InvalidCombination;
+            }
+            var hashes = new List<long>();
+            foreach (var comboString in castCombo.Combo.ToLower().Split('/'))
+            {
+                long hash = ParseCombination(comboString, true);
+                if (hash <= 0)
+                {
+                    return (RegistrationCode)hash;
+                }
+                hashes.Add(hash);
+            }
+            foreach (var hash in hashes)
+            {
+                _comboRegistry.Remove(hash);
+            }
+            _logger.Log("succesfully unregistered " + castCombo.Combo);
+            return RegistrationCode.AllNormal;
+        }
+
+        internal void RegisterGamesBinding(InputCommand binding)
+        {
+            if (_gameBindingRegistry.Contains(binding))
+            {
+                return;
+            }
+            FieldInfo[] fields;
+            if (binding is SingleAxisCommand)
+            {
+                fields = typeof(SingleAxisCommand).GetFields(BindingFlags.NonPublic | BindingFlags.Instance);
+            }
+            else
+            {
+                fields = typeof(DoubleAxisCommand).GetFields(BindingFlags.NonPublic | BindingFlags.Instance);
+            }
```

Comment:
```
 const BindingFlags flags = BindingFlags.NonPublic | BindingFlags.Instance;
var fields = binding is SingleAxisCommand ? typeof(SingleAxisCommand).GetFields(flags) : typeof(DoubleAxisCommand).GetFields(flags);
```
The flags are used elsewhere too, so the constant can be put on the top and reused.

Reply:
done